proxy/src/main/java/com/wavefront/agent/preprocessor/PreprocessorUtil.java

```diff
@@ -261,34 +262,34 @@ private static Predicate processComparisonOp(Map.Entry<String, Object> subElemen
     if (reportableEntity == ReportPoint.class) {
       switch (subElement.getKey()) {
         case "equals":
-          return new ReportPointEqualsPredicate(scope, ruleVal);
+          return new ReportPointEqualsPredicate(scope, (String) ruleVal);
```

Comment:
I think it would be a good idea to check whether ruleVal type is the expected one and throw an exception with a meaningful message, i.e. if "in" is used with String or "equals" used with List. Also add unit tests to verify that these situations are handled correctly.

Reply:
Done.